oauthproxy.go

```diff
@@ -74,7 +74,7 @@ type OAuthProxy struct {
 	allowedRoutes       []allowedRoute
 	redirectURL         *url.URL // the url to receive requests at
 	whitelistDomains    []string
-	provider            providers.Provider
+	provider            []providers.Provider
```

Comment:
Why not a map of ID to provider? Relying on a list an indexing it is inefficient and likely to cause bugs